### DISCUSSION TITLE
Run Reporting

### DISCUSSION THREAD

**Discussion by jonathanmorley** [pnpm/pnpm#5954] [Ideas]
We run `pnpm -r <script name>` in our CI runner. While we can time the execution of this command running for in the CI runner, we do not get any visibility in the per-package time taken to run the script.

Another aspect is that if the execution fails during execution, we do not get visibility into 1) the package that failed, 2) the packages that succeeded, 3) the packages that were cancelled, or 4) the packages that were never started

I would like to propose an argument to `pnpm run`, something like `--report=report.json`, which would output a json file after execution (including failures) with syntax like:

```
{
  "package-1": {
    "status": "failure",
    "duration": 1234
  },
  "package-2": {
    "status": "cancelled",
    "duration": 2345
  },
  "package-3": {
    "status": "queued",
    "duration": undefined
  },
  "package-4": {
    "status": "success",
    "duration": "3456"
  }
}
```

**Comment by zkochan** [comment]
I don't have objections.
